packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,237 @@
+   *
+   * @default - No events.
+   */
+  readonly events?: string[];
+}
+
+/**
+ * Properties for a new notification rule
+ */
+export interface NotificationRuleProps extends NotifyOnEventOptions {
+  /**
+   * The Amazon Resource Name (ARN) of the resource to associate with the notification rule.
+   * Currently, Supported sources include pipelines in AWS CodePipeline and build projects in AWS CodeBuild in this L2 constructor.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-resource
+   */
+  readonly source: INotificationRuleSource;
+}
+
+/**
+ * Represents a notification rule
+ */
+export interface INotificationRule extends IResource {
+
+  /**
+   * The ARN of the notification rule (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   *
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * Adds target to notification rule
+   *
+   * @param target The SNS topic or AWS Chatbot Slack target
+   * @returns boolean - return true if it had any effect
+   */
+  addTarget(target?: INotificationRuleTarget): boolean;
+}
+
+/**
+ * A new notification rule
+ *
+ * @resource AWS::CodeStarNotifications::NotificationRule
+ */
+export class NotificationRule extends Resource implements INotificationRule {
+  /**
+   * Import an existing notification rule provided an ARN
+   * @param scope The parent creating construct
+   * @param id The construct's name
+   * @param notificationRuleArn Notification rule ARN (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   */
+  public static fromNotificationRuleArn(scope: Construct, id: string, notificationRuleArn: string): INotificationRule {
+    class Import extends Resource implements INotificationRule {
+      readonly notificationRuleArn = notificationRuleArn;
+
+      public addTarget(_target?: INotificationRuleTarget) {
+        return false;
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * The source config of notification rule
+   */
+  readonly source: NotificationRuleSourceConfig;
+
+  /**
+   * The target config of notification rule
+   */
+  readonly targets: NotificationRuleTargetConfig[] = [];
+
+  /**
+   * The events of notification rule
+   */
+  readonly events: string[] = [];
+
+  constructor(scope: Construct, id: string, props: NotificationRuleProps) {
+    super(scope, id);
+
+    this.source = props.source.bind(this);
+
+    if (props.events) {
+      this.addEvents(props.events);
+    }
+
+    if (props.target) {
+      this.addTarget(props.target);
+    }
+
+    const resource = new CfnNotificationRule(this, 'Resource', {
+      name: props.notificationRuleName || this.generateName(),
+      status: (props?.enabled === false) ? Status.DISABLED : Status.ENABLED,
```

Comment:
Status is optional, right? Only set it if `props.enabled` is different than `undefined`.

Reply:
How about like this:
```ts
      status: props.enabled !== undefined
        ? !props.enabled ? 'DISABLED' : 'ENABLED'
        : undefined,
```

Reply:
[Status  default value is 'ENABLED'](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-status)

Reply:
> How about like this:
> 
> ```ts
>       status: props.enabled !== undefined
>         ? !props.enabled ? 'DISABLED' : 'ENABLED'
>         : undefined,
> ```

Pretty sure you need some parenthesis there:

```ts
      status: props.enabled !== undefined
        ? (!props.enabled ? 'DISABLED' : 'ENABLED')
        : undefined,
```